webapp/client/src/pages/LoginPage.js

```diff
@@ -1,60 +1,94 @@
 import PropTypes from "prop-types";
 import React from "react";
+import { useTranslation } from "react-i18next";
+import FormFieldIdNr from "../components/FormFieldIdNr";
+import FormFieldUnlockCode from "../components/FormFieldUnlockCode";
 import FormHeader from "../components/FormHeader";
+import FormRowCentered from "../components/FormRowCentered";
 import StepForm from "../components/StepForm";
 import StepHeaderButtons from "../components/StepHeaderButtons";
 
-export default function LoginPage({
-  prevUrl,
-  backLinkText,
-  stepTitle,
-  stepIntro,
-  submitUrl,
-  overviewButton,
-  csrfToken,
-  explanatoryButtonText,
-  nextButtonLabel,
-}) {
+export default function LoginPage({ backLink, stepHeader, form, fields }) {
```

Comment:
Where is the stepHeader coming from? I can't imagine yet how that is set in the template?

Reply:
I've not finished that yet, but my draft looks like this in `unlock_code_activation_steps.py`:

```
def render(self, data, render_info):
        render_info.form.first_field = next(iter(render_info.form))
        return render_template(
            'unlock_code/login.html',
            props=humps.camelize(dict(
                back_link=dict(
                    url=render_info.prev_url,
                    text=render_info.back_link_text if render_info.back_link_text else _('form.back'),
                ),
                step_header=dict(
                    title=render_info.step_title,
                    intro=render_info.step_intro,
                ),
                form=dict(
                    action=render_info.submit_url,
                    csrf_token=generate_csrf(),
                    show_overview_button=bool(render_info.overview_url),
                ),
                fields={
                    field.name: dict(
                        value=field._value(),
                        errors=field.errors,
                    )
                    for field in render_info.form
                },
            )),
```